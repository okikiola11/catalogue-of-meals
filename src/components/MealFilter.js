import React from 'react';
import PropTypes from 'prop-types';

const MealFilter = ({ filterType, filter }) => {
  const categories = [
    'Chicken',
    'Spaghetti',
    'Pizza',
    'Salad',
    'Fish',
  ];

  const handleFilterChange = (e) => {
    filterType(e.target.value);
  };

  return (
    <div>
      <select onChange={handleFilterChange} name="category" defaultValue={filter}>
        {categories.map((category) => (
          <option key={category}>{category}</option>
        ))}
      </select>
    </div>
  );
};

MealFilter.propTypes = {
  filterType: PropTypes.func.isRequired,
  filter: PropTypes.string.isRequired,
};

export default MealFilter;
