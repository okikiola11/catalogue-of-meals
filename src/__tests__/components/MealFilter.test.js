import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen, cleanup } from '@testing-library/react';

import MealFilter from '../../components/MealFilter';

afterEach(cleanup);

it('matches MealFilter snapshot', () => {
  const tree = renderer
    .create(<MealFilter filterType={() => 'mock'} filter="Chicken" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
