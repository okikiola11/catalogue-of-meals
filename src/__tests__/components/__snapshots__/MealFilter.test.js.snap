// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches MealFilter snapshot 1`] = `
<div>
  <select
    className="filter-meal"
    defaultValue="Chicken"
    name="category"
    onChange={[Function]}
  >
    <option>
      Chicken
    </option>
    <option>
      Spaghetti
    </option>
    <option>
      Pizza
    </option>
    <option>
      Beef
    </option>
    <option>
      Fish
    </option>
  </select>
</div>
`;
